src/webview/html/editHistoryHtml.js

```diff
@@ -0,0 +1,105 @@
+  const user = usersById.get(snapshot.user_id);
+  const date = new Date(snapshot.timestamp * 1000);
+  const editedTime = `${humanDate(date)} ${shortTime(date)}`;
+  let displayedContent: string = '';
+  let tag: string = '';
+  let edit_flag;
+
+  if (snapshot.prev_topic === undefined && snapshot.prev_content === undefined) {
+    edit_flag = NOT_EDITED;
+  } else if (snapshot.prev_content !== undefined && snapshot.prev_topic === undefined) {
+    edit_flag = CONTENT_EDITED;
+  } else if (snapshot.prev_content === undefined && snapshot.prev_topic !== undefined) {
+    edit_flag = TOPIC_EDITED;
+  } else {
+    edit_flag = BOTH_EDITED;
+  }
+
+  switch (edit_flag) {
+    case NOT_EDITED: {
+      tag += template`<span class="message-tag">Original message</span>`;
+      displayedContent = snapshot.rendered_content;
+      break;
+    }
+
+    case TOPIC_EDITED: {
+      tag += template`<span class="message-tag">Topic edited by ${
+        user ? user.full_name : 'unknown user'
+      }</span>`;
+
+      // prev_topic is always present for this case. $FlowFixMe.
+      displayedContent = template`Topic:
+      <span class="highlight_text_inserted">${snapshot.topic}</span>
+      <span class="highlight_text_deleted">${snapshot.prev_topic}</span>`;
+      break;
+    }
+
+    case CONTENT_EDITED: {
+      tag += template`<span class="message-tag">Edited by ${
+        user ? user.full_name : 'unknown user'
+      }</span>`;
+
+      // If prev_content is present, then so is content_html_diff. $FlowFixMe.
+      displayedContent = snapshot.content_html_diff;
+      break;
+    }
+
+    case BOTH_EDITED: {
+      tag += template`<span class="message-tag">Edited by ${
+        user ? user.full_name : 'unknown user'
+      }</span>`;
+
+      // prev_topic and content_html_diff are always present for this case. $FlowFixMe.
+      displayedContent = template`Topic:
+      <span class="highlight_text_inserted">${snapshot.topic}</span>
+      <span class="highlight_text_deleted">${snapshot.prev_topic}</span>
+      <br/>$!${snapshot.content_html_diff}`;
+      break;
+    }
+
+    default:
+      logging.error('Invalid edit_flag', { editFlag: edit_flag });
+      break;
+  }
+
+  return template`
+    <div class="edit-history-block">
+      <div class="static-timestamp">${editedTime}</div>
+      <span class="edit-history-content">$!${displayedContent}</span>
+      <div class="message-tags">$!${tag}</div>
+    </div>`;
+};
+
+export default (
+  editHistory: MessageSnapshot[],
+  theme: ThemeName,
+  usersById: Map<number, UserOrBot>,
+  auth: Auth,
+): string => {
+  let html: string = '';
+  for (const snapshot of editHistory) {
+    html += renderSnapshot(snapshot, usersById);
+  }
+
+  return template`
+$!${script(null, auth)}
+$!${css(theme)}
+<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
+<body style="overflow-x: hidden;">
+$!${html}
+</body>`;
```

Comment:
There are some subtle bits here :slightly_smiling_face:. Particularly that `meta` element -- I can only guess at what it does, and would have to look it up.

And this sure has a lot in common with the code in `html.js`! In particular, the subtle bits are exactly the same. (Then the latter has one additional subtle bit, `webkitBugWorkaround`, which is likely unnecessary here but would be harmless.)

Anything that's subtle, I am keen on not duplicating. When code is duplicated, future bugfixes and incidental changes tend to go on just one or the other -- and then it's self-reinforcing: as they diverge, it gets harder to make the same change on both of them or be sure the change would do the same thing on both, so more changes get applied to just the one where the bug happened to be spotted. So because we need this in two places, let's factor it out into one place we can call from both of them.